server/src/main/java/com/cloud/agent/manager/allocator/impl/FirstFitAllocator.java

```diff
@@ -123,11 +138,20 @@
 
         String hostTagOnOffering = offering.getHostTag();
         String hostTagOnTemplate = template.getTemplateTag();
+        String hostTagUefi = "UEFI";
 
         boolean hasSvcOfferingTag = hostTagOnOffering != null ? true : false;
         boolean hasTemplateTag = hostTagOnTemplate != null ? true : false;
 
         List<HostVO> clusterHosts = new ArrayList<HostVO>();
+        List<HostVO> hostsMatchingUefiTag = new ArrayList<HostVO>();
+        if(isVMDeployedWithUefi){
+            hostsMatchingUefiTag = _hostDao.listByHostCapability(type, clusterId, podId, dcId, Host.HOST_UEFI_ENABLE);
+            if (s_logger.isDebugEnabled()) {
+                s_logger.debug("Hosts with tag '" + hostTagUefi + "' are:" + hostsMatchingUefiTag);
+            }
+        }
```

Comment:
please extract, see above
I think it is not good to call this a tag, this already has a wide domain usage that is not meant here. sugest something like "hosts with UEFI capability are: " ...